### PR TITLE
Database gRPC plugins

### PR DESCRIPTION
**Pull request by briankassouf**

### CODE REVIEW

File: builtin/logical/database/dbplugin/grpc_transport.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	// a one second timeout.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	switch c.clientConn.GetState() {
+	case connectivity.Ready, connectivity.Idle:
+	default:
+		return "", ErrPluginShutdown
+	}
+	resp, err := c.client.Type(ctx, &Empty{}, grpc.FailFast(true))
```
Comment (jefferai) [inline builtin/logical/database/dbplugin/grpc_transport.go]: FailFast is actually the default now (see https://github.com/grpc/grpc-go/pull/738 ; it has been for a while, but the earliest grpc code in Vault predated that and using it has just kind of kept going on). Probably can take it out here though.

File: vault/expiration.go [hunk 2]
```diff
@@ -1017,8 +1017,7 @@ func (m *ExpirationManager) revokeEntry(le *leaseEntry) error {
 	}
 
 	// Handle standard revocation via backends
-	resp, err := m.router.Route(logical.RevokeRequest(
-		le.Path, le.Secret, le.Data))
+	resp, err := m.router.Route(logical.RevokeRequest(le.Path, le.Secret, le.Data))
```
Comment (jefferai) [inline vault/expiration.go]: This comes as a non-sequitur :-D

Comment (briankassouf) [reply vault/expiration.go]: Heh, yeah at one point there was a context being passed here, but pulled it out for a different PR

File: plugins/database/cassandra/connection_producer.go [hunk 3]
```diff
@@ -106,15 +107,15 @@ func (c *cassandraConnectionProducer) Initialize(conf map[string]interface{}, ve
...
+		if _, err := c.Connection(ctx); err != nil {
 			return fmt.Errorf("error verifying connection: %s", err)
 		}
 	}
 
 	return nil
 }
 
-func (c *cassandraConnectionProducer) Connection() (interface{}, error) {
+func (c *cassandraConnectionProducer) Connection(ctx context.Context) (interface{}, error) {
```
Comment (calvn) [inline plugins/database/cassandra/connection_producer.go]: Since we are not using `ctx` right now, maybe change it to `_` for the time being.

**Pull request merged**